examples/sim-swap.rs

```diff
@@ -0,0 +1,67 @@
+use std::{error::Error, str::FromStr, sync::Arc};
+
+use ethers::{
+    prelude::abigen,
+    providers::{Http, Middleware, Provider},
+    types::{H160, U256},
+};
+
+use cfmms::pool::UniswapV3Pool;
+
+abigen!(
+    IQuoter,
+r#"[
+    function quoteExactInputSingle(address tokenIn, address tokenOut,uint24 fee, uint256 amountIn, uint160 sqrtPriceLimitX96) external returns (uint256 amountOut)
+]"#;);
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error>> {
+    // load rpc endpoint from local environment
+    let rpc_endpoint = std::env::var("ETHEREUM_MAINNET_ENDPOINT")
+        .expect("Could not get ETHEREUM_MAINNET_ENDPOINT");
```

Comment:
Same as in the `calculate-price` example, lets use `?` everywhere instead of `unwrap()` or `expect()`.